src/coord/src/coord.rs

```diff
@@ -2406,18 +2389,21 @@ where
         let (ops, index_id) = self.generate_view_ops(
             session,
             plan.name,
-            plan.view,
+            plan.view.clone(),
             plan.replace,
             plan.materialize,
         )?;
 
+        let view = plan.view;
+
         match self
-            .catalog_transact(ops, |mut builder, _prev| {
+            .catalog_transact(ops, |mut builder, previous_catalog| {
                 if let Some(index_id) = index_id {
                     if let Some((name, description)) =
-                        Self::prepare_index_build(builder.catalog, &index_id)
+                        Self::prepare_index_build(&builder.catalog, &index_id)
                     {
                         let df = builder.build_index_dataflow(name, index_id, description);
+                        single_materialization::validate(&previous_catalog, &view.depends_on)?;
```

Comment:
Can this be moved above the previous line? It'd avoid building df in the case of an error.